connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/operate/ProcessInstanceContext.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.api.inbound.operate;
+
+import io.camunda.connector.api.inbound.InboundConnectorContext;
+
+/**
+ * Defines the behavior for managing the properties of a process instance within an {@link
+ * InboundConnectorContext}. This interface provides a mechanism for updating the properties
+ * associated with the process instance.
+ */
+public interface ProcessInstanceContext {
+
+  /**
+   * Merges properties from active ProcessInstance variables into the given {@link
+   * InboundConnectorContext}, then evaluates any FEEL expressions found among these properties.
+   *
+   * <p>The method returns true if:
+   *
+   * <ul>
+   *   <li>All properties from active process instances are successfully merged.
+   *   <li>All FEEL expressions among the properties are successfully evaluated.
+   * </ul>
+   *
+   * <p>It returns false if:
+   *
+   * <ul>
+   *   <li>Any FEEL expression among the properties fails to be evaluated.
+   * </ul>
+   *
+   * @param context The inbound connector context whose properties are to be updated and evaluated.
+   * @return true if the properties from active process instances were successfully updated and all
+   *     FEEL expressions evaluated, false otherwise.
+   */
+  boolean updateProperties(InboundConnectorContext context);
```

Comment:
Its somewhat the exact way it was discussed. The Ideas was that this returns all the process intances that are waiting for the message wrapped in a context object that is able to bind properties.